enhancements/proposals/2549-scylladbcluster-backup-repair/README.md

```diff
@@ -0,0 +1,589 @@
+    ScyllaDB Manager API server accepts a float64 value, but overrides any values in (0,1) range. It responds with a 400 status on decimal values > 1. Unfortunately, ScyllaCluster validation allows for decimal values > 1.
+    As we can not make the validation stricter on translation, to maintain backwards compatibility, the translating controller will set an internal annotation `internal.scylla-operator.scylladb.com/scylladb-manager-task-repair-intensity-override` on applied ScyllaDBManagerTask objects.
+    ScyllaDBManagerTask controller will use it to override the value set in the object's specification.
+  - The default value for `FailFast` is removed from the new API. The ScyllaDB Manager server assumes a default value if the parameter is no set explicitly.
+  - The default value for `SmallTableThreshold` is removed from the new API. The ScyllaDB Manager server assumes a default value if the parameter is no set explicitly.
+  - The default value for `Retention` is removed from the new API. The ScyllaDB Manager server assumes a default value if the parameter is no set explicitly.
+  - The default value for `Parallel` is removed from the new API. The ScyllaDB Manager server assumes a default value if the parameter is no set explicitly.
+  - ScyllaDBManagerTask controller will no longer override `NumRetries` when `FailFast` is set to true. If required, such behaviour should be enforced by ScyllaDB Manager server.
+    The translating controller will maintain this behaviour when applying ScyllaDBManagerTasks.
+
+The translating controller will keep respecting the default values defined in ScyllaCluster API and propagate them to applied ScyllaDBManagerTasks.
+
+#### Finalizers
+
+The new ScyllaDBManagerTask controller will set a `scylla-operator.scylladb.com/scylladbmanagertask-protection` finalizer on ScyllaDBManagerTask objects so that the tasks are deleted from ScyllaDB Manager state on object deletion.
+
+#### Status conditions
+
+The new ScyllaDBManagerTask controller will propagate ScyllaDB Manager client errors to `Degraded` status condition.
+
+The translating controller will propagate the error to a corresponding task's status in ScyllaCluster object.
+
+#### Owner references
+
+The translating controller will set the owner references of the created ScyllaDBManagerTask objects to a corresponding ScyllaCluster. The created ScyllaDBManagerTask objects will be garbage collected on ScyllaCluster's deletion according to the deletion policy.
+
+#### Providing credentials for object storage access by ScyllaDB Manager agent sidecar
+
+ScyllaDB Manager agent containers require credentials to the corresponding regional object storage.
+Users will be responsible for configuring the credentials for each datacenter by defining `scylladbclusters.scylla.scylladb.com/v1alpha1.spec.datacenters[].scyllaDBManagerAgent.volumes` and `scylladbclusters.scylla.scylladb.com/v1alpha1.spec.datacenters[].scyllaDBManagerAgent.volumeMounts`.
```

Comment:
(not actionable, just yelling into the void)

the fact that we're speaking in terms of container volumes makes this API a leaky abstraction :cry:. I understand that this is outside of this design's scope and we can't do anything about it in this design doc.

Reply:
> I understand that this is outside of this design's scope and we can't do anything about it in this design doc.

We could but only by heavily interfering with ScyllaDBCluster/ScyllaDBDatacenter API imo, while it was one of the assumptions that we don't touch it for manager-related operations.